coderd/agentapi/stats.go

```diff
@@ -86,16 +87,20 @@ func (a *StatsAPI) UpdateStats(ctx context.Context, req *agentproto.UpdateStatsR
 		}
 		return nil
 	})
-	errGroup.Go(func() error {
-		err := a.Database.UpdateWorkspaceLastUsedAt(ctx, database.UpdateWorkspaceLastUsedAtParams{
-			ID:         workspace.ID,
-			LastUsedAt: now,
+	// NOTE(Cian): This used to be done unconditionally, but now bringing into line with implementation
+	// in coderd/workspaceagents.go
+	if req.Stats.RxBytes > 0 || req.Stats.RxPackets > 0 {
```

Comment:
RxBytes is a counter, not a gauge, so once a connection registers some rx traffic, this will stay non-zero until tailnet stops reporting stats on the connection.

So, in that sense, I think it's functionally equivalent to ConnectionCount (which I want to note _is not equivalent_ to SessionCount, which only measures SSH).